graphql_compiler/compiler/emit_cypher.py

```diff
@@ -68,14 +75,23 @@ def _emit_code_from_cypher_step(cypher_step):
             u'%(left_edge_mark)s-[:%(edge_type)s%(quantifier)s]-%(right_edge_mark)s' +
             step_vertex_pattern
         )
-        linked_location_name, _ = cypher_step.linked_location.get_location_name()
+        if isinstance(cypher_step.linked_location, FoldScopeLocation):
```

Comment:
Perhaps a good opportunity for a helper function here? This code exists at least twice, once around line 55 and once here.

Reply:
These are slightly different since the condition for which we want the full path name is slightly different. Earlier it depends on if the current step is in a fold scope, but here we want the linked location and if this is the first step in a fold scope, the linked location would be a Location.

I'm open to passing in the cypher_step and the boolean for whether or not to use the `FoldScopeLocation` function though since the logic besides that is the same.

Reply:
I'm not sure I understand. What I was envisioning was something like:

```python
def new_helper_function_with_better_name(location):
    if isinstance(location, FoldScopeLocation):
        return fold_scope_location_full_path_name(step_location)
    elif isinstance(location, Location):
        location_name, _ = location.get_location_name()
        return location_name

    raise AssertionError(<unreachable code error message>)
```

Then, around line 59-66 you'd do
```python
step_location_name = new_helper_function_with_better_name(cypher_step.as_block.location)
```
and around lines 85-91 you'd do
```python
linked_location_name = new_helper_function_with_better_name(cypher_step.linked_location)
```

Is this not equivalent to what we have here?

Reply:
ah, you're right-- the fact that I checked if it was a fold step earlier threw me off